__tests__/healthcheck.test.js

```diff
@@ -0,0 +1,368 @@
+        status: 'fail',
+        description: 'Wallet-headless health',
+        checks: {
+          'Wallet health_wallet': [
+            {
+              componentName: 'Wallet health_wallet',
+              componentType: 'internal',
+              status: 'fail',
+              output: 'Wallet is not ready. Current state: Syncing',
+              time: expect.any(String),
+            },
+          ],
+          'Fullnode http://fakehost:8083/v1a/': [
+            {
+              componentName: 'Fullnode http://fakehost:8083/v1a/',
+              componentType: 'fullnode',
+              status: 'pass',
+              output: 'Fullnode is responding',
+              time: expect.any(String),
+            },
+          ],
+          'TxMiningService http://fake.txmining:8084/': [
+            {
+              componentName: 'TxMiningService http://fake.txmining:8084/',
+              componentType: 'service',
+              status: 'pass',
+              output: 'Tx Mining Service is healthy',
+              time: expect.any(String),
+            },
+          ],
+        }
+      });
+
+      wallet.isReady = originalIsReady;
+      wallet.state = originalState;
+    });
+
+    it('should return 503 when the fullnode is not healthy', async () => {
+      TestUtils.httpMock.onGet('/version').reply(503, { status: 'fail' });
+
+      const response = await TestUtils.request
+        .get('/health');
+      expect(response.status).toBe(503);
+      expect(response.body.status).toBe('fail');
+      expect(response.body.description).toBe('Wallet-headless health');
+
+      expect(response.body.checks['Wallet health_wallet'][0].componentName).toBe('Wallet health_wallet');
+      expect(response.body.checks['Wallet health_wallet'][0].componentType).toBe('internal');
+      expect(response.body.checks['Wallet health_wallet'][0].output).toBe('Wallet is ready');
+      expect(response.body.checks['Wallet health_wallet'][0].status).toBe('pass');
+
+      expect(response.body.checks['Fullnode http://fakehost:8083/v1a/'][0].componentName).toBe('Fullnode http://fakehost:8083/v1a/');
+      expect(response.body.checks['Fullnode http://fakehost:8083/v1a/'][0].componentType).toBe('fullnode');
+      expect(response.body.checks['Fullnode http://fakehost:8083/v1a/'][0].status).toBe('fail');
+      expect(response.body.checks['Fullnode http://fakehost:8083/v1a/'][0].output).toBe('Fullnode reported as unhealthy: {"status":"fail"}');
+
+      expect(response.body.checks['TxMiningService http://fake.txmining:8084/'][0].componentName).toBe('TxMiningService http://fake.txmining:8084/');
+      expect(response.body.checks['TxMiningService http://fake.txmining:8084/'][0].componentType).toBe('service');
+      expect(response.body.checks['TxMiningService http://fake.txmining:8084/'][0].status).toBe('pass');
+      expect(response.body.checks['TxMiningService http://fake.txmining:8084/'][0].output).toBe('Tx Mining Service is healthy');
```

Comment:
I think this whole section is now redundant and could be removed safely

Reply:
Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/348/commits/c105cbff68d1a174a5bb6e323ade85f80b5d213d